src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -2348,6 +2348,12 @@ void G1CollectedHeap::par_iterate_regions_array(HeapRegionClosure* cl,
 
 HeapWord* G1CollectedHeap::block_start(const void* addr) const {
   HeapRegion* hr = heap_region_containing(addr);
+  // The CollectedHeap API requires us to not fail for any given address within
+  // the heap. HeapRegion::block_start() has been optimized to not accept addresses
+  // outside of the allocated area.
+  if (addr >= hr->top()) {
+    return hr->top();
```

Comment:
In the caller, `block_is_obj` from `if (p != NULL && CollectedHeapT::heap()->block_is_obj(p)) {` would choke on `top` as well, right? Returning `nullptr` makes more sense to me.

Reply:
Or fixing `block_is_obj`...

Reply:
Originally the method returned `nullptr`, but I'm not sure this is the best option.

Reply:
I just return `nullptr`, fixing `block_is_obj` (returning false for addresses >= top) does not seem to be worth.